tests/global_communication/test.cpp

```diff
@@ -28,12 +38,44 @@ int main(int argc, char **argv) {
     // now add our custom printer
     listeners.Append(new mpi_listener("results_global_communication"));
 
-    // record the local return value for tests run on this mpi rank
-    //      0 : success
-    //      1 : failure
-    auto result = RUN_ALL_TESTS();
+    int return_value = 0;
+    try {
+        auto arg = argv+1;
+        while (*arg) {
+            if (auto comm_size = to::parse_opt<unsigned>(arg, 'd', "dryrun")) {
+                if (*comm_size==0) {
+                    throw to::parse_opt_error(*arg, "must be positive integer");
+                }
+                // Note that this must be set again for each test that uses a different
+                // number of cells per domain, e.g.
+                //      policy::set_sizes(policy::size(), new_cells_per_rank)
+                policy::set_sizes(*comm_size, 0);
+            }
+            else if (auto o = to::parse_opt(arg, 'h', "help")) {
+                to::usage(argv[0], usage_str);
+                return 0;
+            }
+            else {
+                throw to::parse_opt_error(*arg, "unrecognized option");
+            }
+        }
+
+        // record the local return value for tests run on this mpi rank
+        //      0 : success
+        //      1 : failure
+        return_value = RUN_ALL_TESTS();
+    }
+
+    catch (to::parse_opt_error& e) {
```

Comment:
If these were the only exceptions that were being thrown, I would generally agree, even though I think it is six of one, half a dozen of the other.

However, the calls to `to::parse_opt` can also throw a `to::parse_opt_error` exception. So throwing the same exception type actually simplifies the error handling here, because you need the try-catch block anyway.